godtools/App/Share/Data/UserAuthentication/AuthenticationProviders/AppleAuthentication+AuthenticationProviderInterface.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  AppleAuthentication+AuthenticationProviderInterface.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 5/16/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SocialAuthentication
+import Combine
+
+extension AppleAuthentication: AuthenticationProviderInterface {
+    func getPersistedAccessToken() -> AuthenticationProviderAccessToken? {
+        
+        // TODO: - access token renewal will come through MobileContentAPI?
+        
+        return nil
+    }
+    
+    func authenticatePublisher(presentingViewController: UIViewController) -> AnyPublisher<AuthenticationProviderAccessToken?, Error> {
+        
+        return authenticatePublisher()
+            .map { (response: AppleAuthenticationResponse) in
+                
+                guard
+                    let idToken = response.identityToken,
+                    let givenName = response.fullName?.givenName,
+                    let familyName = response.fullName?.familyName
+                else {
+                    return nil
+                }
+                                
+                return AuthenticationProviderAccessToken.apple(
+                    idToken: idToken,
+                    givenName: givenName,
+                    familyName: familyName
+                )
+            }
+            .eraseToAnyPublisher()
+    }
+    
+    func renewAccessTokenPublisher() -> AnyPublisher<AuthenticationProviderAccessToken, Error> {
+        
+        // TODO: - access token renewal will come through MobileContentAPI?
+        
+        return Just(AuthenticationProviderAccessToken.apple(idToken: "", givenName: "", familyName: ""))
+            .setFailureType(to: Error.self)
+            .eraseToAnyPublisher()
+    }
+    
+    func signOutPublisher() -> AnyPublisher<Void, Error> {
+        
+        return Just(()).setFailureType(to: Error.self)
+            .eraseToAnyPublisher()
+    }
+    
+    func getAuthUserPublisher() -> AnyPublisher<AuthUserDomainModel?, Error> {
+        
+        // TODO: Should return an AuthUserDomainModel if user info can be obtained from AppleAuthentication. Implement in GT-2027. ~Levi
```

Comment:
For this we will need to return the AuthUserDomainModel.  I realize now we will need some additional work in social-authentication-ios so we can get user info for AuthUserDomainModel.

Google and Facebook SDKs will persist data for a current authenticated user such as access/refresh tokens, user info, etc.  See GoogleSignIn Framework Reference for ```GIDGoogleUser``` https://developers.google.com/identity/sign-in/ios/reference/Classes/GIDGoogleUser

It would be nice if we could do something similar for AppleAuthentication.swift this way we can fetch a current user and add the email, first name, and last name to the AuthUserDomainModel.